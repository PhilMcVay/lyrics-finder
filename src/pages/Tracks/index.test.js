import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen, waitForElement } from "@testing-library/react";
import axiosMock from "axios";

import TracksPage from "./index";

const mockArtistData = {
  data: {
    artists: [{ id: "1234", name: "The Police" }],
  },
};

const mockWorksData = {
  data: {
    works: [
      { title: "C Song", type: "Song" },
      { title: "Video 1", type: "Video" },
      { title: "B Song", type: "Song" },
      { title: "A Song", type: "Song" },
    ],
  },
};

describe("<TracksPage>", () => {
  it("Should Render", () => {
    render(<TracksPage />);

    expect(screen.getByTestId("tracks-page")).toBeInTheDocument();
  });

  it("Should fetch the list of tracks for an artist and display them in alphabetical order", async () => {
    axiosMock.get
      .mockResolvedValueOnce(mockArtistData)
      .mockResolvedValueOnce(mockWorksData);

    render(
      <BrowserRouter>
        <TracksPage />
      </BrowserRouter>
    );

    // Display loading state while fetching data
    expect(screen.getByText("Loading Tracks...")).toBeInTheDocument();

    const trackList = await waitForElement(() => screen.getByText("A Song"));

    // Displays Artist Name
    expect(screen.getByText("The Police")).toBeInTheDocument();

    // Display all songs
    expect(screen.getByText("A Song")).toBeInTheDocument();
    expect(screen.getByText("B Song")).toBeInTheDocument();
    expect(screen.getByText("C Song")).toBeInTheDocument();

    // Display list in alphabetical order
    expect(trackList).toMatchSnapshot();
  });
});
