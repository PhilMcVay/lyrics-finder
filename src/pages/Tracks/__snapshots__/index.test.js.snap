// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TracksPage> Should fetch the list of tracks for an artist and display them in alphabetical order 1`] = `
<ul
  class="sc-gKAblj dgCImu"
  data-testid="tracks-list"
>
  <li
    class="sc-eCApGN dyTrPi"
  >
    <a
      href="/The Police/A Song"
    >
      A Song
    </a>
  </li>
  <li
    class="sc-eCApGN dyTrPi"
  >
    <a
      href="/The Police/B Song"
    >
      B Song
    </a>
  </li>
  <li
    class="sc-eCApGN dyTrPi"
  >
    <a
      href="/The Police/C Song"
    >
      C Song
    </a>
  </li>
</ul>
`;
